### ISSUE TITLE
Reserve and UserReserve balances are wrong 

### ISSUE THREAD

**Issue by ronlv10** [aave/protocol-subgraphs#82]
## TLDR
In v3, there are a bunch of wallets that have wrong balances in the subgraph compared to the on-chain data. 

## Example
For example, [0xbc1631afcb916bda28af42955fc970bf004596f8](https://polygonscan.com/address/0xbc1631afcb916bda28af42955fc970bf004596f8) on Polygon has 0 on-chain balance: 
![image](https://user-images.githubusercontent.com/17575462/187254021-5081b77e-02cc-4165-9984-4bdd35505f6e.png) but, querying this user from the subgraph and using [aave-utilities](https://github.com/aave/aave-utilities) formatter, the user has [balances](https://pastebin.com/crLMsxtT). 
Same goes for wallets that still have dust balance on chain: [0xee2826453a4fd5afeb7ceffeef3ffa2320081268](https://snowtrace.io/address/0xee2826453a4fd5afeb7ceffeef3ffa2320081268) on Avalanche : 0.00000018 WETH.e on chain- comparing to [1+ WETH.e](https://pastebin.com/4Z97fBe1) using the subgraph.

## Root cause - account balance computation updated in v3
It seems like the `Burn` and `Mint` events were changed in AAVE v3. In v3, The event value reports the amount that should be minted/burned while taking into account the interest which was accumulated (which impact the amount of mint/burned) since the last update. 
In v2, the events reported the amount which was supplied/withdrawn, and the interest could be calculated using other fields in the subgraph like `liquidityIndex`, `liquidityRate`, etc.

### Scope of Issue

- Any wallet that executed two actions with a delta such that interest was accrued stores the wrong balance.
    - For example, a wallet that has performed the `supply` method twice or a combination of `withdraw` and `supply`.
- This bug also affects any field that is affected by interest accumulated by the wallet balances. Namely, this effects Reserve metrics such as `totalSupply` and `totalBorrow` per asset.

### Impact

- This is a large-scale bug as it essentially affects all wallets (barring wallets that have only ever supplied).
- This means that anyone consuming data on wallets/health/risk from the officially supported subgraphs was reading the wrong data.
- Any teams, traders, or institutions attempting to make data-driven decisions based on subgraph data are affected.

## Proposed Fix
In order to keep consistency and backward compatibility to the structure of v2 (and to aave-utilities functions),  we need to sum up the `Burn.value` and `Burn.balanceIncrease` and use this value for all the calculations. The opposite goes for Mint - we need to subtract  `Mint.balanceIncrease` from `Mint.value`.

## Proposed PR
**We have created a PR that fixes the issue, but we couldn't submit it because of permission to the repository. If you can provide me access to create PRs, we would be happy to contribute to the repo with the fix PR.**

**Comment by Marykassman** [comment]
That’s my contract address been taken over in the case of not paying for the address approval and I could like anyone get back to me on what to do or what you need .

**Comment by defispartan** [comment]
> That’s my contract address been taken over in the case of not paying for the address approval and I could like anyone get back to me on what to do or what you need .

I have no idea what this means or how this relates to this issue

**Comment by defispartan** [comment]
@ronlv10 Thank you so much for this contribution and the detailed write up. We are testing this change in a staging deployment and will have it out in prod shortly after some testing

I believe that a similar issue exists for the V2 subgraphs as well. I'm not sure if you would have any capacity to research this as well? If not it's totally fine, we can pick it up after finishing this change.

**Comment by ronlv10** [comment]
Hey @defispartan. I am really happy I could help. 
I attached below the subgraphs I deployed using this fix, in case you want to try them out.

> I believe that a similar issue exists for the V2 subgraphs as well. I'm not sure if you would have any capacity to research this as well? If not it's totally fine, we can pick it up after finishing this change.

I would be happy to try to help. I think I need some more context regarding the issue in V2: IIUC, the change in the event types was only on v3, no? Any examples of the issue in V2 would be great.



Subgraph links:
[Arbitrum Subgraph](https://thegraph.com/hosted-service/subgraph/ronlv10/aave-v3-arbitrum)
[Harmony Subgraph](https://thegraph.com/hosted-service/subgraph/ronlv10/aave-v3-harmony)
[Polygon Subgraph](https://thegraph.com/hosted-service/subgraph/ronlv10/aave-v3-polygon)
[Fantom Subgraph](https://thegraph.com/hosted-service/subgraph/ronlv10/aave-v3-fantom)
[Avalanche Subgraph](https://thegraph.com/hosted-service/subgraph/ronlv10/aave-v3-avalanche)
[Optimism Subgraph](https://thegraph.com/hosted-service/subgraph/ronlv10/aave-v3-optimism-fix)

**Comment by defispartan** [comment]
Thank you so much for your help! We just deployed and merged an update including your change to all production and testnet subgraphs 🚀 

And yeah we can keep this thread open and share more details about the bugs that we're seeing in V2 as well
